File: sdks/sdkwork-control-plane-sdk/sdkwork-control-plane-sdk-flutter/composed/lib/src/nodes_module.dart
import 'package:control_plane_backend_sdk/control_plane_backend_sdk.dart';

import 'context.dart';

class ControlPlaneNodesModule {
  final ControlPlaneSdkContext context;

  ControlPlaneNodesModule(this.context);

  Future<JsonObject> activate(Identifier nodeId) {
    return context.backendClient.nodes.activateNode(nodeId);
  }

  Future<JsonObject> drain(Identifier nodeId) {
    return context.backendClient.nodes.drainNode(nodeId);
  }

  Future<JsonObject> migrateRoutes(Identifier nodeId, JsonObject body) {
    return context.backendClient.nodes.migrateNodeRoutes(nodeId, body);
  }
}

File: sdks/sdkwork-control-plane-sdk/sdkwork-control-plane-sdk-typescript/composed/src/nodes-module.ts
import type { Identifier, JsonObject } from './types.js';
import type { ControlPlaneSdkContext } from './sdk-context.js';

export class ControlPlaneNodesModule {
  constructor(private readonly context: ControlPlaneSdkContext) {}

  activate(nodeId: Identifier): Promise<JsonObject> {
    return this.context.backendClient.nodes.activateNode(nodeId);
  }

  drain(nodeId: Identifier): Promise<JsonObject> {
    return this.context.backendClient.nodes.drainNode(nodeId);
  }

  migrateRoutes(nodeId: Identifier, body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.nodes.migrateNodeRoutes(nodeId, body);
  }
}
